tests/unit/test_data.py

```diff
@@ -9,3 +15,44 @@ def test_repeating_loader():
         assert next(loader) == 1
         assert next(loader) == 2
         assert next(loader) == 3
+
+
+@pytest.mark.parametrize('train_batch_size, drop_last',
+                         [(1,
+                           True),
+                          (4,
+                           True),
+                          (1,
+                           False),
+                          (4,
+                           False)])
+def test_dataloader_drop_last(tmpdir, train_batch_size, drop_last):
+    config_dict = {
```

Comment:
I am so sorry that I did not communicate clearly previously. I am sorry for the extra work. 
Let me try again and hopefully. 

1. `config_dict `should be like 
```python
{ 'train_batch_size': train_batch_size, 'dataloader_drop_last': drop_last, ....}
```
This will allow `train_batch_size `and `dataloader_drop_last `to pass into `deepspeed.initialize()` normally via `args`. 

2. `train_dataset `should be passed directly to deepspeed.initialize() as [training_data](https://github.com/microsoft/DeepSpeed/blob/336dd089e5e11d46daf3abfad048d71b72912504/deepspeed/__init__.py#L66) arg. This will validate that all the changes you made in the library are working correctly. The returned [training_data_loader](https://github.com/microsoft/DeepSpeed/blob/336dd089e5e11d46daf3abfad048d71b72912504/deepspeed/__init__.py#L159) value should be used in the rest of unit test. Apologies that I previously suggested that you should use `DeepSpeedDataLoader` directly in the unit test, it is not necessary. 

Please let me know of anything that is not clear. Thanks.

Reply:
Great question. [args_from_dict()](https://github.com/microsoft/DeepSpeed/blob/ddffbae021fef24bf64945d45ace9bac7e08508d/tests/unit/simple_model.py#L230) converts the `config_dict` to a json file and sets `args.deepspeed_config` to the file path. This is how fields in `config_dict` get passed into the DeepSpeed engine. 

The final bit of code looks good.

Reply:
One more thing, it seems you will need to enable fp16 in config_dict, like [this](https://github.com/microsoft/DeepSpeed/blob/ddffbae021fef24bf64945d45ace9bac7e08508d/tests/unit/test_fp16.py#L317-L319).

Reply:
Also, is it possible to run the unit tests on your box following these [guide](https://github.com/microsoft/DeepSpeed/blob/master/CONTRIBUTING.md#unit-tests)?

Reply:
@tjruwase 

Done!

Reply:
Cool. Thanks. I will launch the CI again on this side.